pkg/release-controller/release_info.go

```diff
@@ -793,73 +832,75 @@ func (r *ExecReleaseInfo) GetIssuesWithDemoLink(issues []string) (result []jiraB
 	return result, err
 }
 
-func (r *ExecReleaseInfo) GetRemoteLinksWithConcurrency(issues []string) (result map[string][]string, err error) {
-	// This will prevent a Panic if/when the release-controller's are run without the necessary jira flags
+func (r *ExecReleaseInfo) GetRemoteLinksWithConcurrency(issues []string, requestInterval time.Duration) (map[string][]string, error) {
+	// Prevent Panic if Jira client is not set
 	if r.jiraClient == nil {
-		return result, fmt.Errorf("unable to communicate with Jira")
+		return nil, fmt.Errorf("unable to communicate with Jira")
 	}
 
 	var (
 		mapIssueDemoLink = make(map[string][]string)
 		wg               sync.WaitGroup
 		mu               sync.Mutex
-		maxWorkers       = 10
-		workers          = make(chan struct{}, maxWorkers)
-		buf              bytes.Buffer
+		workers          = make(chan struct{}, 5) // it does not make sense to have more workers since the API rate is limited
+		ticker           = time.NewTicker(requestInterval)
+		errorBuilder     strings.Builder
+		demoRegex        = regexp.MustCompile(`\bdemo\b`) // Precompile regex
 	)
 
+	defer ticker.Stop()
+
 	worker := func(issue string) {
 		defer wg.Done()
-		defer func() { <-workers }()
+		<-ticker.C // Wait for API call slot
 
-		links, err := r.jiraClient.GetRemoteLinks(issue)
-		if err != nil {
-			mu.Lock()
-			err = fmt.Errorf("search failed: %w", err)
-			buf.WriteString(err.Error() + "\n")
-			mu.Unlock()
-			return
+		var links []jiraBaseClient.RemoteLink
+		if cachedLinks, found := r.jiraCache.Get(fmt.Sprintf("%s_remote_link", issue)); found {
```

Comment:
Since `fmt.Sprintf("%s_remote_link", issue)` is used multiple times, in this function, I'd extract it into a variable and reference that instead of building the same string multiple times.

Reply:
Fixed